From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Sun, 24 Sep 2023 20:57:02 +0100
Subject: [PATCH] [sailfishos][gecko] Restore nsAppShell.h

The widget/qt/nsAppShell.h file is missing from the ESR 91 source trea.
This change restores it from the ESR78 source tree.
---
 widget/qt/nsAppShell.h | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)
 create mode 100644 widget/qt/nsAppShell.h

diff --git a/widget/qt/nsAppShell.h b/widget/qt/nsAppShell.h
new file mode 100644
index 000000000000..2d2f37d75864
--- /dev/null
+++ b/widget/qt/nsAppShell.h
@@ -0,0 +1,37 @@
+/* -*- Mode: c++; tab-width: 2; indent-tabs-mode: nil; -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef nsAppShell_h__
+#define nsAppShell_h__
+
+#include <qsocketnotifier.h>
+#include "nsBaseAppShell.h"
+#include "nsCOMPtr.h"
+
+/**
+ * Native QT Application shell wrapper
+ */
+
+class nsAppShell : public QObject,
+                   public nsBaseAppShell
+{
+  Q_OBJECT
+
+public:
+  nsAppShell() { };
+
+  nsresult Init();
+
+  virtual bool event (QEvent *e);
+
+protected:
+  virtual void ScheduleNativeEventCallback();
+  virtual bool ProcessNextNativeEvent(bool mayWait);
+  virtual ~nsAppShell();
+};
+
+
+#endif // nsAppShell_h__
+
